tests/TestCase/Command/BakeMigrationDiffCommandTest.php

```diff
@@ -201,6 +201,71 @@ public function testBakingDiffWithAutoIdIncompatibleUnsignedPrimaryKeys(): void
         $this->runDiffBakingTest('WithAutoIdIncompatibleUnsignedPrimaryKeys');
     }
 
+    /**
+     * Tests that baking a diff with --plugin option only includes tables with Table classes
+     */
+    public function testBakingDiffWithPluginOnlyIncludesTablesWithTableClasses(): void
+    {
+        $this->skipIf(!env('DB_URL_COMPARE'));
+
+        // Create some test tables in the comparison database
+        $connection = ConnectionManager::get('test_comparisons');
+
+        // Create a table that has a Table class in the TestBlog plugin
+        $connection->execute('CREATE TABLE IF NOT EXISTS articles (
+            id INT AUTO_INCREMENT PRIMARY KEY,
+            title VARCHAR(255)
+        )');
+
+        // Create a table that does NOT have a Table class in the TestBlog plugin
+        $connection->execute('CREATE TABLE IF NOT EXISTS orphan_table (
+            id INT AUTO_INCREMENT PRIMARY KEY,
+            name VARCHAR(255)
+        )');
+
+        // Create plugin migration history table
+        $connection->execute('CREATE TABLE IF NOT EXISTS test_blog_phinxlog (
+            version BIGINT NOT NULL,
+            migration_name VARCHAR(100) DEFAULT NULL,
+            start_time TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+            end_time TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+            breakpoint TINYINT(1) NOT NULL DEFAULT 0,
+            PRIMARY KEY (version)
+        )');
```

Comment:
This SQL isn't going to work across platforms. You'll either need to generalize it quite a bit, or make this test conditional for MySQL.